tiny_cnn/core/backend_nnp.h

```diff
@@ -87,29 +92,25 @@ class nnp_backend : public backend {
     void deconv2d(cnn_size_t                 index,
                   const std::vector<vec_t*>& in_data,
                   std::vector<vec_t*>&       out_data) {
-        if (!params_d_->has_bias) {
+        /*if (!params_d_->has_bias) {
             throw nn_error("NNPACK Convolution requires a bias term.");
         }
 
         if (params_d_->w_stride != 1 || params_d_->h_stride != 1) {
             throw nn_error("NNPACK Convolution requires stride 1.");
         }
 
-#ifdef CNN_USE_NNPACK
         deconv_layer_worker_specific_storage& cws =
             (*deconv_layer_worker_storage_)[index];
-        copy_and_pad_delta(cws.curr_delta_padded, *in_grad[0]);
+        copy_and_pad_delta(cws.curr_delta_padded, *in_grad[0]); //  *in_grad[0] does not exist here!
         const vec_t& W    = *in_data[1];
```

Comment:
@Wangyida Are you sure that you compiled this? There are variables that are not declared